src/ui/screen/screen_setup_mods.cpp

```diff
@@ -0,0 +1,281 @@
+}
+
+void CScreenSetupMods::CreateInterface()
+{
+    CWindow*        pw;
+    CLabel*         pl;
+    CButton*        pb;
+    CList*          pli;
+    Math::Point     pos, ddim;
+    std::string     name;
+
+    CScreenSetup::CreateInterface();
+    pw = static_cast<CWindow*>(m_interface->SearchControl(EVENT_WINDOW5));
+    if ( pw == nullptr )  return;
+
+    // Displays a list of unloaded mods:
+    pos.x = ox+sx*3;
+    pos.y = oy+sy*9;
+    ddim.x = dim.x*6;
+    ddim.y = dim.y*1;
+    GetResource(RES_TEXT, RT_MODS_UNLOADED, name);
+    pl = pw->CreateLabel(pos, ddim, 0, EVENT_LABEL11, name);
+    pl->SetTextAlign(Gfx::TEXT_ALIGN_LEFT);
+
+    pos.y = oy+sy*3.75f;
+    ddim.x = dim.x*6.5f;
+    ddim.y = dim.y*6.05f;
+    pli = pw->CreateList(pos, ddim, 0, EVENT_INTERFACE_MODS_UNLOADED);
+    pli->SetState(STATE_SHADOW);
+    UpdateUnloadedModList();
+
+    // Displays a list of loaded mods:
+    pos.x = ox+sx*9.5f;
+    pos.y = oy+sy*9;
+    ddim.x = dim.x*6;
+    ddim.y = dim.y*1;
+    GetResource(RES_TEXT, RT_MODS_LOADED, name);
+    pl = pw->CreateLabel(pos, ddim, 0, EVENT_LABEL12, name);
+    pl->SetTextAlign(Gfx::TEXT_ALIGN_LEFT);
+
+    pos.y = oy+sy*3.75f;
+    ddim.x = dim.x*6.5f;
+    ddim.y = dim.y*6.05f;
+    pli = pw->CreateList(pos, ddim, 0, EVENT_INTERFACE_MODS_LOADED);
+    pli->SetState(STATE_SHADOW);
+    UpdateLoadedModList();
+
+    pos = pli->GetPos();
+    ddim = pli->GetDim();
+    pos.x  = ox+sx*8.2f;
+    pos.y  = oy+sy*2;
+    ddim.x = dim.x*1;
+    ddim.y = dim.y*1;
+    pb = pw->CreateButton(pos, ddim, 40, EVENT_INTERFACE_WORKSHOP);
+    pb->SetState(STATE_SHADOW);
+
+    pos.x += dim.x*1.3f;
+    pb = pw->CreateButton(pos, ddim, 57, EVENT_INTERFACE_MODS_DIR);
+    pb->SetState(STATE_SHADOW);
+
+    pos.x += dim.x*1.3f;
+    ddim.x = dim.x*2.5f;
+    pb = pw->CreateButton(pos, ddim, -1, EVENT_INTERFACE_LOAD);
+    pb->SetState(STATE_SHADOW);
+    pb->ClearState(STATE_ENABLE);
+
+    pos.x += dim.x*2.8f;
+    pb = pw->CreateButton(pos, ddim, -1, EVENT_INTERFACE_UNLOAD);
+    pb->SetState(STATE_SHADOW);
+    pb->ClearState(STATE_ENABLE);
+}
+bool CScreenSetupMods::EventProcess(const Event &event)
+{
+    CWindow*               pw;
+    CButton*               pb;
+    CList*                 pl;
+    int                    result;
+    std::string            modName, modPath, modPathRaw, OFF = "~";
+
+    if (!CScreenSetup::EventProcess(event)) return false;
+
+    pw = static_cast<CWindow*>(m_interface->SearchControl(EVENT_WINDOW5));
+    if ( pw == nullptr )  return false;
+
+    if (event.type == EVENT_INTERFACE_LOAD)
+    {
+        pl = static_cast<CList*>(pw->SearchControl(EVENT_INTERFACE_MODS_UNLOADED));
+        if ( pl == nullptr )  return false;
+        modName = pl->GetItemName(pl->GetSelect());
+
+        modPathRaw = CResourceManager::GetSaveLocation() + "/" + "mods" + "/";
+        modPath = modPathRaw.c_str();
+        boost::filesystem::rename(modPath+OFF+modName, modPath+modName);
```

Comment:
UI classes should not contain any logic other than the display code. Move all of the enable/disable/mod listing/check if enabled logic to a new CModManager class.